grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcHealthCheckServiceBuilder.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.healthcheck.HealthCheckUpdateListener;
+import com.linecorp.armeria.server.healthcheck.ListenableHealthChecker;
+
+/**
+ * Builds a {@link GrpcHealthCheckService}.
+ */
+public final class GrpcHealthCheckServiceBuilder {
+
+    private final ImmutableSet.Builder<ListenableHealthChecker> healthCheckers = ImmutableSet.builder();
+
+    private final ImmutableMap.Builder<String, ListenableHealthChecker> grpcHealthCheckers =
+            ImmutableMap.builder();
+
+    private final ImmutableList.Builder<HealthCheckUpdateListener> updateListeners = ImmutableList.builder();
+
+    GrpcHealthCheckServiceBuilder() {}
+
+    /**
+     * Adds the specified {@link ListenableHealthChecker}s that determine the healthiness of the {@link Server}.
+     */
+    public GrpcHealthCheckServiceBuilder checkers(ListenableHealthChecker... healthCheckers) {
+        return checkers(ImmutableSet.copyOf(requireNonNull(healthCheckers, "healthCheckers")));
+    }
+
+    /**
+     * Adds the specified {@link ListenableHealthChecker}s that determine the healthiness of the {@link Server}.
+     */
+    public GrpcHealthCheckServiceBuilder checkers(Iterable<? extends ListenableHealthChecker> healthCheckers) {
+        this.healthCheckers.addAll(requireNonNull(healthCheckers, "healthCheckers"));
+        return this;
+    }
+
+    /**
+     * Adds the pair of the service name and the {@link ListenableHealthChecker} that determine
+     * the healthiness of the specified gRPC service.
+     * The specified health checker is also used for determining the healthiness of the {@link Server}.
```

Comment:
Because the `healthChecker` is for a service, I think it shouldn't affect the healthiness of the server.
There might be a chance that only one service has trouble but other services work normally. What do you think?

Reply:
It was a difficult decision, but this is what I chose.

> Because the `healthChecker `is for a service, I think it shouldn't affect the healthiness of the server.

However, if we focus on this opinion, I think you are right.
I will fix it.